### PR TITLE
Timesheet summary report and reminder mail who haven't filled timesheet 

### PR DESCRIPTION
**Pull request by vaibhav8186**

### CODE REVIEW

File: app/controllers/time_sheets_controller.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class TimeSheetsController < ApplicationController
+  class TimeSheetsController < ApplicationController
```
[the remark this answers is not in the document]
Comment (vaibhav8186) [reply app/controllers/time_sheets_controller.rb]: Done this

File: app/controllers/time_sheets_controller.rb [hunk 2]
```diff
@@ -141,15 +141,23 @@ def individual_project_report
...
+    end
+
     respond_to do |format|
       format.html
-      format.csv {
-        filename = @project.name + '_' + @from_date.to_date.strftime('%b') + '_' + @from_date.to_date.strftime('%Y') + '.csv'
-        report = TimeSheet.create_project_report_in_csv(@project, @from_date.to_date, @to_date.to_date)
-        send_data report, filename: filename
+      format.xlsx{
+         response.headers['Content-Disposition'] = 'attachment; filename = "Timesheet_summary_report.xlsx"'
```
[the remark this answers is not in the document]
Comment (vaibhav8186) [reply app/controllers/time_sheets_controller.rb]: Done

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -150,7 +150,7 @@ def build_addresses
   end
 
   def load_emails_and_projects
-    @emails = User.approved.collect(&:email)
+    @emails = User.or({role: "HR"},{role: "Manager"},{role: "Admin"}).pluck(:email)
```
[the remark this answers is not in the document]
Comment (vaibhav8186) [reply app/controllers/users_controller.rb]: created scope - management_team

File: app/mailers/weekly_timesheet_report_mailer.rb [hunk 4]
```diff
@@ -7,4 +7,14 @@ def send_weekly_timesheet_report(csv, email, unfilled_time_sheet_report)
     attachments["weekly_timesheet_report_#{Date.today}.csv"] = csv
     mail(subject: 'Weekly timesheet report', to: [email] + DEFAULT_TIMESHEET_MANAGERS)
   end
+
+  def send_report_who_havent_filled_timesheet( options = {} )
+    @text = options[:text]
+    attachments["Employees- Not Filled Timesheet.csv"] = options[:csv]
+    mail(
+      subject:"Employees List-  who haven't filled timesheet",
```
[the remark this answers is not in the document]
Comment (vaibhav8186) [reply app/mailers/weekly_timesheet_report_mailer.rb]: In the mail description, from_date and to_date is mentioned. Still, I have mentioned in the subject also.

File: spec/models/time_sheet_spec.rb [hunk 5]
```diff
@@ -1942,6 +1946,65 @@
...
+        :user => user, :project => project )
+      leave = FactoryGirl.create(:leave_application, start_at: Date.today - 2, end_at: Date.today,
+        user: user, leave_status: LEAVE_STATUS[1])
+      from_date = Date.today - 3
+      to_date   = Date.today
+      TimeSheet.get_users_who_not_filled_timesheet(from_date, to_date)
+      expect(ActionMailer::Base.deliveries.count).to eq(0)
+    end
+
+    it "Do not send mail if user not assigned on any project project" do
```
[the remark this answers is not in the document]
Comment (vaibhav8186) [reply spec/models/time_sheet_spec.rb]: Done

File: spec/models/time_sheet_spec.rb [hunk 6]
```diff
@@ -1942,6 +1946,65 @@
       project      = FactoryGirl.create(:project)
       user_project = FactoryGirl.create(:user_project, user: user, project: project)
       timesheet    = FactoryGirl.create(:time_sheet, user: user, project: project)
+    end
+  end
+  
+  context "User without time_sheet" do
```
[the remark this answers is not in the document]
Comment (vaibhav8186) [reply spec/models/time_sheet_spec.rb]: These specs are related to only one feature that's why these specs are clubbed into one context.

File: app/mailers/weekly_timesheet_report_mailer.rb [hunk 7]
```diff
@@ -7,4 +7,14 @@ def send_weekly_timesheet_report(csv, email, unfilled_time_sheet_report)
     attachments["weekly_timesheet_report_#{Date.today}.csv"] = csv
     mail(subject: 'Weekly timesheet report', to: [email] + DEFAULT_TIMESHEET_MANAGERS)
```
Comment (swapnilchincholkar) [inline app/mailers/weekly_timesheet_report_mailer.rb]: @vaibhav8186 Should we need to uniq the combination of email ids here?

**Pull request merged**